casper/src/main/scala/coop/rchain/casper/SafetyOracle.scala

```diff
@@ -73,27 +73,39 @@ sealed abstract class SafetyOracleInstances {
-                                        candidateMetadata
-                                      )
-          _ <- Span[F].mark("agreeing-validator-to-weight")
-          maxCliqueWeight <- if (2L * agreeingValidatorToWeight.values.sum < totalWeight) {
-                              0L.pure[F]
-                            } else {
-                              agreementGraphMaxCliqueWeight(
-                                blockDag,
-                                candidateMetadata,
-                                agreeingValidatorToWeight
-                              )
-                            }
-          _ <- Span[F].mark("max-clique-weight")
-
-          faultTolerance = 2 * maxCliqueWeight - totalWeight
-        } yield faultTolerance.toFloat / totalWeight
+          _                      <- Log[F].debug(s"Calculating faultTolearance of block ${candidateBlockHash} ")
+          maybeCandidateMetadata <- blockDag.lookup(candidateBlockHash)
+          faultTolerance <- maybeCandidateMetadata match {
+                             case Some(candidateMetadata) =>
+                               for {
+                                 totalWeight <- computeTotalWeight(blockDag, candidateMetadata)
+                                 _           <- Span[F].mark("total-weight")
+                                 agreeingValidatorToWeight <- computeAgreeingValidatorToWeight(
+                                                               blockDag,
+                                                               candidateMetadata
+                                                             )
+                                 _ <- Span[F].mark("agreeing-validator-to-weight")
+                                 maxCliqueWeight <- if (2L * agreeingValidatorToWeight.values.sum < totalWeight) {
+                                                     0L.pure[F]
+                                                   } else {
+                                                     agreementGraphMaxCliqueWeight(
+                                                       blockDag,
+                                                       candidateMetadata,
+                                                       agreeingValidatorToWeight
+                                                     )
+                                                   }
+                                 _ <- Span[F].mark("max-clique-weight")
+
+                                 faultTolerance = 2 * maxCliqueWeight - totalWeight
+                               } yield faultTolerance.toFloat / totalWeight
+                             // if the node can to find the block, it would return -1 and regard that block as orphaned
+                             case None =>
+                               Log[F].info(
+                                 s"calculate faultTolerance blockHash ${candidateBlockHash} failed because it can not be found in store."
+                               ) >> (-1L).toFloat
```

Comment:
@zsluedem just to be clear what I meant by ambiguity - this will return valid `show-block` with faultToleranceThreshold=-1 for block that is not added yet to the DAG (previously we had error in this case).